### ISSUE TITLE
AArch64: Failure in cmdLineTester_jvmtitests_hcr_3

### ISSUE THREAD

**Issue by knn-k** [eclipse/openj9#9017]
Failure link
------------
https://ci.eclipse.org/openj9/job/Test_openjdk11_j9_sanity.functional_aarch64_linux_xl_Release/1/
rc007 and ta001 in cmdLineTester_jvmtitests_hcr_3

```
[2020-03-29T02:09:50.064Z] Testing: rc007
[2020-03-29T02:09:50.064Z] Test start time: 2020/03/29 02:09:48 Coordinated Universal Time
[2020-03-29T02:09:50.064Z] Running command: "/home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_aarch64_linux_xl_Release/openjdkbinary/j2sdk-image/bin/java" -Xnocompressedrefs -Xgcpolicy:gencon -Xjit:count=0 -Xnocompressedrefs  -Xdump    -agentlib:jvmtitest=test:rc007 -cp "/home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_aarch64_linux_xl_Release/openjdk-tests/TKG/../../jvmtest/functional/cmdLineTests/jvmtitests/jvmtitest.jar:/home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_aarch64_linux_xl_Release/openjdk-tests/TKG/../TKG/lib/asm-all.jar" com.ibm.jvmti.tests.util.TestRunner
[2020-03-29T02:09:50.064Z] Time spent starting: 4 milliseconds
[2020-03-29T02:09:55.835Z] Time spent executing: 6652 milliseconds
[2020-03-29T02:09:55.835Z] Test result: FAILED
[2020-03-29T02:09:55.835Z] Output from test:
[2020-03-29T02:09:55.835Z]  [OUT] *** Testing [1/3]:	testMultipleClassRedefinesToNewVersions
[2020-03-29T02:09:55.835Z]  [OUT] *** Test took 134 milliseconds
[2020-03-29T02:09:55.835Z]  [OUT] OK
[2020-03-29T02:09:55.835Z]  [OUT] 
[2020-03-29T02:09:55.835Z]  [OUT] *** Testing [2/3]:	testMultipleClassRedefinesToThemselves
[2020-03-29T02:09:55.835Z]  [OUT] *** Test took 16 milliseconds
[2020-03-29T02:09:55.835Z]  [OUT] FAILED
[2020-03-29T02:09:55.835Z]  [OUT] 
[2020-03-29T02:09:55.835Z]  [OUT] *** Testing [3/3]:	testMultipleRedefinesFromSameHierarchy
[2020-03-29T02:09:55.835Z]  [OUT] *** Test took 29 milliseconds
[2020-03-29T02:09:55.835Z]  [OUT] OK
[2020-03-29T02:09:55.835Z]  [OUT] 
[2020-03-29T02:09:55.835Z] >> Success condition was not found: [Return code: 0]
```
```
[2020-03-29T02:13:52.483Z] Testing: ta001
[2020-03-29T02:13:52.483Z] Test start time: 2020/03/29 02:13:52 Coordinated Universal Time
[2020-03-29T02:13:52.483Z] Running command: "/home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_aarch64_linux_xl_Release/openjdkbinary/j2sdk-image/bin/java" -Xnocompressedrefs -Xgcpolicy:gencon -Xjit:count=0 -Xnocompressedrefs  -Xdump    -agentlib:jvmtitest=test:ta001 -cp "/home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_aarch64_linux_xl_Release/openjdk-tests/TKG/../../jvmtest/functional/cmdLineTests/jvmtitests/jvmtitest.jar:/home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_aarch64_linux_xl_Release/openjdk-tests/TKG/../TKG/lib/asm-all.jar" com.ibm.jvmti.tests.util.TestRunner
[2020-03-29T02:13:52.483Z] Time spent starting: 4 milliseconds
[2020-03-29T02:14:01.327Z] Time spent executing: 9074 milliseconds
[2020-03-29T02:14:01.327Z] Test result: FAILED
[2020-03-29T02:14:01.327Z] Output from test:
[2020-03-29T02:14:01.327Z]  [OUT] *** Testing [1/13]:	testBCIUsingASM_InjectNPELogic
[2020-03-29T02:14:01.327Z]  [OUT] Pre  Replace: 1
[2020-03-29T02:14:01.327Z]  [OUT] Expected NullPointerException received
[2020-03-29T02:14:01.327Z]  [OUT] *** Test took 323 milliseconds
[2020-03-29T02:14:01.327Z]  [OUT] OK
(... lines omitted ...)
[2020-03-29T02:14:01.327Z]  [OUT] 
[2020-03-29T02:14:01.327Z]  [OUT] *** Testing [6/13]:	testBCIUsingASM_inject_CatchWithSelfGOTO
[2020-03-29T02:14:01.327Z]  [ERR] 
[2020-03-29T02:14:01.327Z]  [ERR] 
[2020-03-29T02:14:01.327Z]  [ERR] *** Invalid JIT return address 0000FFFF9C303D70 in 0000FFFF9C201D08
[2020-03-29T02:14:01.327Z]  [ERR] 
[2020-03-29T02:14:01.327Z]  [ERR] 02:13:58.965 0xffff9c201a00    j9vm.249    *   ** ASSERTION FAILED ** at swalk.c:1565: ((0 ))
```

**Comment by knn-k** [comment]
`cmdLineTester_jvmtitests_hcr_openi9_none_SCC_3` fails in the same way as `cmdLineTester_jvmtitests_hcr_3`.

**Comment by knn-k** [comment]
The compressed refs build skips these tests because they use because the `-Xnocompressedrefs` option.

**Comment by Akira1Saitoh** [comment]
`rc007` fails with x86 vm if `optLevel=cold` is set:
```
root@c06e2be1802b:~/work# ./jdk-11.0.6+10-jre/bin/java -version                                                                           openjdk version "11.0.6" 2020-01-14
OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.6+10)
Eclipse OpenJ9 VM AdoptOpenJDK (build openj9-0.18.1, JRE 11 Linux amd64-64-Bit 20200122_403 (JIT enabled, AOT enabled)
OpenJ9   - 51a5857d2
OMR      - 7a1b0239a
JCL      - da35e0c380 based on jdk-11.0.6+10)
```

```
root@c06e2be1802b:~/work# ./jdk-11.0.6+10-jre/bin/java -Xgcpolicy:gencon -Xjit:count=0,optLevel=cold -Xnocompressedrefs -Xdump -agentlib:jvmtitest=test:rc007 -cp jvmtitest.jar:asm-all.jar com.ibm.jvmti.tests.util.TestRunner
*** Testing [1/3]:      testMultipleClassRedefinesToNewVersions
*** Test took 54 milliseconds
OK

*** Testing [2/3]:      testMultipleClassRedefinesToThemselves
*** Test took 8 milliseconds
FAILED

*** Testing [3/3]:      testMultipleRedefinesFromSameHierarchy
*** Test took 20 milliseconds
OK
```

If `optLevel` is not specified, the test passes with x86 vm:
```
root@c06e2be1802b:~/work# ./jdk-11.0.6+10-jre/bin/java -Xgcpolicy:gencon -Xjit:count=0 -Xnocompressedrefs -Xdump -agentlib:jvmtitest=test:rc007 -cp jvmtitest.jar:asm-all.jar com.ibm.jvmti.tests.util.TestRunner
*** Testing [1/3]:      testMultipleClassRedefinesToNewVersions
*** Test took 89 milliseconds
OK

*** Testing [2/3]:      testMultipleClassRedefinesToThemselves
*** Test took 10 milliseconds
OK

*** Testing [3/3]:      testMultipleRedefinesFromSameHierarchy
*** Test took 15 milliseconds
OK
```

**Comment by Akira1Saitoh** [comment]
With debug build of `aarch64` vm, `rc007` fails with assertion:
```
*** Testing [1/3]:      testMultipleClassRedefinesToNewVersions
Assertion failed at /root/openj9-openjdk-jdk11/build/linux-aarch64-normal-server-release/vm/compiler/../compiler/control/HookedByTheJit.cpp:2996: 0
        JIT HCR should make all methods recompilable, so startPC=0x7f31020d34 should have a persistentBodyInfo
#0: /home/jetson/work/openj9/jdk-xl/lib/default/libj9jit29.so(+0x73d454) [0x7f87cf9454]
#1: /home/jetson/work/openj9/jdk-xl/lib/default/libj9jit29.so(+0x748cd0) [0x7f87d04cd0]
#2: /home/jetson/work/openj9/jdk-xl/lib/default/libj9jit29.so(+0x43a500) [0x7f879f6500]
#3: /home/jetson/work/openj9/jdk-xl/lib/default/libj9jit29.so(+0x43a7f0) [0x7f879f67f0]
#4: /home/jetson/work/openj9/jdk-xl/lib/default/libj9jit29.so(+0x12f6f4) [0x7f876eb6f4]
#5: /home/jetson/work/openj9/jdk-xl/lib/default/libj9jvmti29.so(+0x31af8) [0x7f8c41aaf8]
```

**Comment by Akira1Saitoh** [comment]
Same assertion hit for `ta001`:
```
*** Testing [1/13]:     testBCIUsingASM_InjectNPELogic
Assertion failed at /root/openj9-openjdk-jdk11/build/linux-aarch64-normal-server-release/vm/compiler/../compiler/control/HookedByTheJit.cpp:2996: 0
        JIT HCR should make all methods recompilable, so startPC=0x7f54d81cdc should have a persistentBodyInfo
#0: /home/jetson/work/openj9/jdk-xl/lib/default/libj9jit29.so(+0x73d454) [0x7fabcf9454]
#1: /home/jetson/work/openj9/jdk-xl/lib/default/libj9jit29.so(+0x748cd0) [0x7fabd04cd0]
#2: /home/jetson/work/openj9/jdk-xl/lib/default/libj9jit29.so(+0x43a500) [0x7fab9f6500]
#3: /home/jetson/work/openj9/jdk-xl/lib/default/libj9jit29.so(+0x43a7f0) [0x7fab9f67f0]
#4: /home/jetson/work/openj9/jdk-xl/lib/default/libj9jit29.so(+0x12f6f4) [0x7fab6eb6f4]
#5: /home/jetson/work/openj9/jdk-xl/lib/default/libj9jvmti29.so(+0x31af8) [0x7fb00a9af8]
```

**Comment by pshipton** [comment]
https://ci.eclipse.org/openj9/view/Test/job/Grinder/765/

**Comment by knn-k** [comment]
Similar issue, but different? #9024

**Comment by knn-k** [comment]
rc018 and ta001 fails in walking the stack with the latest XL build.

**Comment by knn-k** [comment]
Only ta001 fails in https://ci.eclipse.org/openj9/job/Test_openjdk11_j9_sanity.functional_aarch64_linux_xl_Personal_testList_0/1/tapResults/

**Comment by knn-k** [comment]
Fixed by #9953.  Closing.
https://ci.eclipse.org/openj9/job/Test_openjdk11_j9_sanity.functional_aarch64_linux_xl_Personal_testList_0/2/tapResults/
